generator/vulkan/c_parse.zig

```diff
@@ -263,6 +263,7 @@ pub fn parseMember(allocator: Allocator, xctok: *XmlCTokenizer, ptrs_optional: b
         .field_type = decl.decl_type,
         .bits = null,
         .is_buffer_len = false,
+        .is_optional = false,
```

Comment:
This made the most sense to me I think, as we need to track whether the spec considers certain fields optional or not. It's a little redundant with pointer optionalness, though.

Reply:
Hm yes its kind of weird that the optionalness is duplicated here. I guess one alternative would be to add a new type in `TypeInfo`, though with this the types from `registry.zig` would have the same exact representation as the parse tree anymore so that is also not great. I think leaving it like this is fine.